packages/packagers/js/src/ScopeHoistingPackager.js

```diff
@@ -53,6 +53,7 @@ const GLOBALS_BY_CONTEXT = {
     ...Object.keys(globals.node),
     ...Object.keys(globals.browser),
   ]),
+  edge: new Set([...BUILTINS, ...Object.keys(globals.worker)]),
```

Comment:
I think using `globals.workers` may not be correct because it has: https://github.com/sindresorhus/globals/blob/main/globals.json#L1129-L1135
And `self: true` that edge workers may not have implemented.

Reply:
The only effect this has is that Parcel doesn't not create some `var onclose = ...` variable (if it decides to try that for some reason), so that if there's a global of that name, it will rename it to `var _onclose = ...` etc...

https://github.com/parcel-bundler/parcel/blob/9e2d5d0d60d08d65b5ae6cd765c907a8753bbf39/packages/packagers/js/src/ScopeHoistingPackager.js#L351-L355

In that sense, I thought that workers is close enough here (and a conservative choice).

Reply:
Do we need to account for other globals in https://edge-runtime.vercel.app/features/available-apis, https://proposal-common-min-api.deno.dev/#index, or other runtimes somehow?

Reply:
https://developers.cloudflare.com/workers/runtime-apis/web-standards/#:~:text=Cloudflare%20Workers%20uses%20the%20V8,with%20no%20need%20for%20transpilers.

@devongovett I think the tricky thing to support them in every custom runtime(wrappers on top of barebone js engine) is they include different subsets of web APIs into their runtime based on the usage. 

For more generic runtimes like `deno`, they may provide a larger amount of default globals.

But for some specialised runtimes with very thin layers on top of v8(https://github.com/denoland/rusty_v8) or isolated-vm(https://github.com/laverdet/isolated-vm). The build target of these kinds of runtimes will have nothing. All the compatibility layers will be hand-written by the runtime creators.

I think the purpose of this `edge` target is for environments that only contain: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects assuming the runtime is a pure js engine(v8/jscore/...etc).

Reply:
> Do we need to account for other globals in https://edge-runtime.vercel.app/features/available-apis, https://proposal-common-min-api.deno.dev/#index, or other runtimes somehow?

AFAICT, these are all subsets of the browser globals, which is why I chose the worker globals.


> I think the tricky thing to support them in every custom runtime(wrappers on top of barebone js engine) is they include different subsets of web APIs into their runtime based on the usage.

As I've said, the worst thing that can happen here is a variable being called `var _ReadableStreamBYOBReader` as opposed to `var ReadableStreamBYOBReader` even if `ReadableStreamBYOBReader` is not available in the runtime. This is **not** about polyfilling the `URLSearchParams` global for runtimes not supporting it.

And as far I can tell, all callers of `getTopLevelName` do `this.getTopLevelName("$"+foo+"$"+bar);` anyway, so I'm not even sure why we have these globals in the first place... @devongovett ?

Reply:
hmm I think maybe vestiges of an older implementation where we had exports that weren't prefixed.